### ISSUE TITLE
rationale behind this project

### ISSUE THREAD

**Issue by dominikandreas** [bbernhard/signal-cli-rest-api#3]
Hi Bernhard, 

thanks for making signal available to home-assistant! I really appreciate it.

I was wondering why you chose to create this rest-api wrapper around the signal-cli instead of directly calling signal-cli from home-assistant. It seems to me like a bit of an unnecessary overhead, so I was wondering if there is some rationale behind all of this that I'm missing and would be pleased if you could elaborate on this.

Thanks,
Dominik

**Comment by bbernhard** [comment]
Hi Dominik,

the main reasons for creating this `signal-cli` wrapper were: 

* I've multiple services in my local network that can send alerts via signal. So having one central entry point for sending messages makes life easier for me (it reduces the maintenance effort significantly). e.g I am using the signal REST API to send messages via the Android tasker app (of course I could also do that through the Home Assistant REST API interface, but I find the Home Assistant REST API a bit cumbersome to use; but that's just my personal opinion). 

* It allows to run the REST API on a different system than Home Assistant. 

* Initially, I indeed tried to use the `signal-cli` commandline tool directly with Home Assistant, but unfortunately I couldn't make it work. I spent a few hours debugging the issue and in the end it turned out to be a `musl` problem. Afaik the Home Assistant docker container is based on an alpine docker base image, which uses `musl` instead of `glibc`. And for some (unknown) reason the alpine java runtime took really long (> 30sec.) to start the  `signal-cli` program...so sending a signal message was slow as hell. 
I've tried to install `glibc` inside the Home Assistant docker image to circumvent the issue, but this turned out to be more work than I've anticipated, created other problems and would've been a maintenance nightmare.
